@Y-T-G ok I added a submodule import test but it's failing here. Normally this works:

```python
import numpy.random
random_number = numpy.random.rand()
```

Test is showing:

```python
with lazy():
    import numpy.random
random_number = numpy.random.rand()
    
Traceback (most recent call last):
  File "<input>", line 3, in <module>
AttributeError: 'builtin_function_or_method' object has no attribute 'rand'
```

--- a/autoimport/main.py
+++ b/autoimport/main.py
@@ -41,10 +41,11 @@ class LazyLoader(types.ModuleType):
 
 class lazy:
     """Context manager for lazy imports."""
-    _lazy_modules = {}  # Store lazy modules here
+
     def __init__(self):
         """Initializes a context manager for lazy module imports to optimize startup time and memory usage."""
         self._original_import = builtins.__import__
+        self._lazy_modules = {}  # Store lazy modules here
         self._globals = {}
 
     def __enter__(self):
@@ -71,10 +72,10 @@ class lazy:
                 return types.SimpleNamespace(
                     **{from_item: self._lazy_modules[f"{module_name}.{from_item}"] for from_item in fromlist}
                 )
-            elif module_name in globals:
-                return globals[module_name]
             else:
                 if module_name in sys.modules:
+                    self._lazy_modules[module_name] = globals[module_name]  # avoid duplicate issue
+                elif module_name in sys.modules:
                     self._lazy_modules[module_name] =  sys.modules[module_name]  # module already loaded in session
                 elif module_name not in self._lazy_modules:
                     self._lazy_modules[module_name] = LazyLoader(module_name)

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -65,7 +65,8 @@ dependencies = []
 # Optional dependencies ------------------------------------------------------------------------------------------------
 [project.optional-dependencies]
 dev = [
-    "numpy",  # for import tests
+    "numpy",  # for tests
+    "torch",
 ]
 
 [project.urls]  # Optional

--- a/tests/test_autoimport.py
+++ b/tests/test_autoimport.py
@@ -22,13 +22,20 @@ class TestLazyImports(unittest.TestCase):
         result = json.dumps({"test": 123})
         self.assertIsInstance(result, str)
 
-    def test_submodule_imports(self):
-        """Test nested module imports and functionality."""
+    def test_submodule_imports_numpy(self):
+        """Test numpy module imports and functionality."""
         with lazy():
             import numpy.random
         random_number = numpy.random.rand()
         self.assertLess(random_number, 1.0)
 
+    def test_submodule_imports_torch(self):
+        """Test torch module imports and functionality."""
+        with lazy():
+            import torch.nn
+        module = torch.nn.Identity()
+        self.assertIsInstance(module, torch.nn.Identity)
+    
     def test_direct_lazyloader(self):
         """Test direct LazyLoader instantiation."""
         base64_lazy = LazyLoader("base64")

